File: 01. DataStructures/02. Stack/LeetCode Questions/03. Build an Array With Stack Operations #1441/Solution.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        ArrayList<String> arr = new ArrayList<String>();
        int input = 1; //1,2,3,4,....,n
        for(int i=0; i<target.length; i++) {
            while(input<target[i]) {
                arr.add("Push");
                arr.add("Pop");
                input++;
            }
            arr.add("Push");
            input++;
        }
        return arr;
    }
}

File: 01. DataStructures/02. Stack/LeetCode Questions/03. Build an Array With Stack Operations #1441/Solution.js
var buildArray = function(target, n) {
    
    const outputArr = [];
    const mySet = new Set();
    
    for(let i = 0; i < target.length; i++) {
        mySet.add(target[i]);
    }
    j = 1;
    while(j <= n && mySet.size) {
        
        outputArr.push('Push');
        
        if(!mySet.has(j)) {
            outputArr.push('Pop');
        }
        
        if(mySet.has(j)) {
            mySet.delete(j);
        }  
        j++;
    }
    
    return outputArr;
};

console.log(buildArray([1,3], 3));
